firefly/main.py

```diff
@@ -17,22 +21,40 @@ def load_function(function_spec):
     mod_name, func_name = function_spec.rsplit(".", 1)
     mod = importlib.import_module(mod_name)
     func = getattr(mod, func_name)
-    return (func_name, func)
+    return ("/"+func_name, func_name, func)
 
 def load_functions(function_specs):
     return [load_function(function_spec) for function_spec in function_specs]
 
+def parse_config_file(config_file):
+    if not os.path.exists(config_file):
+        raise FireflyError("Specified config file does not exist.")
+    with open(config_file) as f:
+        config_dict = yaml.safe_load(f)
+    return config_dict
+
+def parse_config_data(config_dict):
+    return [(spec["path"], func_name, load_function(spec["function"])[2])
+            for func_name, spec in config_dict["functions"].items()]
```

Comment:
`load_function(spec["function"])[2]` looks ugly and confusing.

How about:

    [load_function(f['function'], name=name, path=f['path']) for name, f in config_dict["functions"].items()]

and make the load_function take optional name and path arguments?